packages/backend/src/apps/vault-workspace/actions/get-table-data/get-data-out-metadata.ts

```diff
@@ -0,0 +1,36 @@
+import { IDataOutMetadata, IExecutionStep } from '@plumber/types'
+
+import { VAULT_ID } from '../../common/constants'
+
+async function getDataOutMetadata(
+  step: IExecutionStep,
+): Promise<IDataOutMetadata> {
+  const data = step.dataOut
+  if (!data) {
+    return null
+  }
+  const _metadata = data?._metadata as Record<string, any>
```

Comment:
nit: I think technically more correct to type this as `IJSONObject` instead of `Record<string, any>`